web3/src/main/java/org/hiero/mirror/web3/service/ContractStateServiceImpl.java

```diff
@@ -83,35 +83,36 @@ public Optional<byte[]> findStorageByBlockTimestamp(
      * @return slotKey-value pairs for contractId
      */
     private Optional<byte[]> findStorageBatch(final EntityId contractId, final byte[] key) {
-        final var contractSlotsCache = this.contractSlotsCache.get(
-                contractId, () -> cacheManagerSlotsPerContract.getCache(contractId.toString()));
-        contractSlotsCache.putIfAbsent(ByteBuffer.wrap(key), EMPTY_VALUE);
-
+        final var contractSlotsCache = ((CaffeineCache) this.contractSlotsCache.get(
+                contractId, () -> cacheManagerSlotsPerContract.getCache(contractId.toString())));
         // Cached slot keys for contract, whose slot values are not present in the contractStateCache
         final var cachedSlots = new ArrayList<byte[]>();
-        ((CaffeineCache) contractSlotsCache).getNativeCache().asMap().keySet().forEach(slot -> {
+        contractSlotsCache.putIfAbsent(ByteBuffer.wrap(key), EMPTY_VALUE);
```

Comment:
`cachedSlots` has to be a `HashSet<ByteBuffer>` for `contains` to work, also for efficient `contains` check

```suggestion
        final var wrappedkey = ByteBuffer.wrap(key);
        contractSlotsCache.putIfAbsent(wrappedkey, EMPTY_VALUE);
        final var cachedSlots = contractSlotsCache.getNativeCache().asMap().keySet();
        ...
        
        boolean isKeyEvictedFromCache = !cachedSlots.contains(wrappedKey);
```

Reply:
I think If we do the check in the first loop the performance won’t change, considering the Caffeine cache is using ConcurrentHashMap as an underlying structure. Since it is unordered the check via iteration would has O(n/2) complexity on average similar to the ArrayList contains(). Also the lambda expression would become more complex.
On the other hand,  the contains() method  of the HashSet would have constant time complexity on average compared to O(n) in the ArrayList case.

Reply:
The `contains` in ConcurrentHashMap should be also constant.

Reply:
Agree we can also do the `isKeyEvictedFromCache` check inside the loop. We can't avoid the loop because `contractStateRepository.findStorageBatch` takes `List<byte[]>` while the keyset in the underlying `cache.asMap()` is a `Set<ByteBuffer>`.